scripts/release.sh

```diff
@@ -119,6 +118,23 @@ main() {
     exit 1
   fi
 
+  # Check that gh is installed and logged in.
+  log_callout "Check gh installation"
+  if ! command -v gh >/dev/null; then
+    log_error "Cannot find gh. Please follow the installation instructions at https://github.com/cli/cli#installation"
+    exit 1
+  fi
+  local gh_version
+  gh_version=$(gh --version | head -n1 | awk 'match($0,/([0-9]+\.){2}[0-9]+/) {print substr($0,RSTART,RLENGTH)}')
+  if [ "$(cut -d. -f1 <(echo "${gh_version}"))" -lt "2" ] || [ "$(cut -d. -f2 <(echo "${gh_version}"))" -lt "57" ]; then
+    log_error "The minimum version required for gh is 2.57.0, got ${gh_version}"
+    exit 1
+  fi
+  if [ "${DRY_RUN}" != "true" ] && ! gh auth status &>/dev/null; then
```

Comment:
Should we validate `gh auth status` irrespective of `DRY_RUN`?  If I was using `DRY_RUN=true` I would still want to know that I had a problem with `gh`.

Reply:
I avoided the check because it would fail when running on the CI. One option would be to add a new execution flag (i.e., `--no-github-release`) so we can set that in the release test and still keep it green. WDYT?

Reply:
Makes sense, having a new flag we can use in CI sounds good to me.